content/4_2_rethink_db.tex

```diff
@@ -0,0 +1,412 @@
+
+Additionally change feeds can be squashed. Squashed feeds buffer change data and trigger change commands less frequent. This greatly improves performance on the client and server.
+
+\subsection{Reflection}
+
+\subsubsection{Cap Theorem}
+In general RethinkDB does value Consistency over Availability (see Listing \autoref{fig:rethink_cap_theorem}). The database guarantees that a write operation is distributed to all systems before the change can be read, like described in \autoref{sec:rethink_arch_query_execution}. Therefore any normal read operation will always return the cluster wide consensus at any give time.\\
+
+The normal behaviour can be changed on a per query basis. A user can specify that the given request should favour availability. In such a case the database returns the next available record regardless of it being on a primary replica. A query using this behaviour can be created as follows
+
+\mint{js}{r.table("Houses").run({readMode: "outdated"});}
+
+\begin{figure}[ht]
+    \centering
+    \includegraphics{img/rethinkCapTheorem.png}
+    \caption{Overview of the CAP Theorem in RethinkDB}
+    \label{fig:rethink_cap_theorem}
+\end{figure}
+
+\subsubsection{Performance}
+
+In a test conducted by the RethinkDB core team the developers concluded that the database scales horizontally in a near linear fashion \autocite{rethinkdb:performance} as can be seen in \autoref{fig:rethink_rethink_performance}. The developers further imply that there will be ongoing efforts to improve upon this results.
+
+\begin{figure}[ht]
+    \includegraphics[width=.5\linewidth]{img/rethinkperf_rw.png} \quad
+    \includegraphics[width=0.5\linewidth]{img/rethinkperf_rsync.png}
+    \\[\baselineskip]
+    \includegraphics[width=0.5\linewidth]{img/rethinkperf_rasync.png} \quad
+    \includegraphics[width=0.5\linewidth]{img/rethinkperf_aquery.png}
+    \caption{Overview of performance test results\autocite{rethinkdb:performance}}
+    \label{fig:rethink_rethink_performance}
+\end{figure}
+
+\subsubsection{Comparison to real-time sync APIs}
+The functionality of RethinkDB can be compared to many real-time sync API services like Pusher\footnote{https://pusher.com} or Firebase\footnote{https://firebase.google.com/products/realtime-database/}. There are some key differences between those kinds of systems and RethinkDB.
+
+The capability to store information is the most noticeable difference. Sync APIs don't store any data. They are intended to send uniform event streams to a large amount of recipients. Other key differences can be taken from \autoref{tbl:rethink_realtimesync}.
+
+\begin{table}[ht]
+\caption{Comparison of RethinkDB and real-time sync APIs}
+\label{tbl:rethink_realtimesync}
+\centering
+\begin{tabular}{|p{0.6\textwidth}|p{0.4\textwidth}|}
```

Comment:
With spacing around the table, it is too large for the page. Use something like `0.55` and `0.35` for column widths.